File: test/_harnesses/typescript/readme_reader.ts
import * as fs from 'fs';

import * as AppMessages from '../../../out/generated/typescript/AppMessages';

const data = fs.readFileSync("./vec3.msg");
const dv = new DataView(new Uint8Array(data).buffer);
const msg = AppMessages.Vector3Message.FromBytes(dv, 0).val;
if (msg.y == Math.fround(4096.1234)) {
    console.log("Ready to go!");
}

File: test/_harnesses/go/readme_reader/readme_reader.go
package main

import (
	"os"

	"messages/app_messages"
)

func main() {
	dat, _ := os.Open("./vec3.msg")
	defer dat.Close()
	msg := app_messages.Vector3MessageFromBytes(dat)
	if msg.X == 1.0 && msg.Y == 4096.1234 && msg.Z < 0.0 {
		print("Ready to go!\n")
	}
}
